graphene/relay/connection.py

```diff
@@ -108,10 +111,13 @@ def type(self):
         assert issubclass(connection_type, Connection), (
             '{} type have to be a subclass of Connection. Received "{}".'
         ).format(self.__class__.__name__, connection_type)
-        return connection_type
+        return type
 
     @classmethod
     def resolve_connection(cls, connection_type, args, resolved):
+        if isinstance(connection_type, NonNull):
```

Comment:
This will check the `connection_type` each time the field is resolved, however we can move this logic to `connection_resolver` so is just executed once :)

Reply:
Done!